docs/products/kafka/concepts/upgrade-procedure.rst

```diff
@@ -0,0 +1,253 @@
+Apache Kafka® upgrade procedure
+===============================
+
+One of the benefits of using a managed service like Aiven for Apache Kafka® is the automated upgrade procedure. 
+
+The upgrade procedure is executed during:
+    
+* maintenance updates
+* plan changes
+* cloud region migrations
+* manual node replacements performed by an Aiven operator
+    
+All the above operations involve creating new broker nodes to replace existing ones.
+
+Upgrade procedure steps
+---------------------------
+
+The following set of steps are executed during an upgrade procedure, the example shows a possible scenario for a 3-node Apache Kafka service.
+
+.. mermaid::
+
+    flowchart TD;
+        subgraph KafkaCluster
+            subgraph Node3
+                PartitionB2([Partition B])
+                PartitionC2([Partition C])
+            end
+            subgraph Node2
+                PartitionA2([Partition A])
+                PartitionC1([Partition C])
+            end
+            subgraph Node1
+                PartitionA1([Partition A])
+                PartitionB1([Partition B])
+            end
+        end
+        
+    
+
+1. New Apache Kafka® nodes are started alongside the existing nodes
```

Comment:
on my screen the diagram for the new nodes looks too tiny. Maybe better to do it a line below?
<img width="1439" alt="Screenshot 2022-08-03 at 11 43 09" src="https://user-images.githubusercontent.com/4600541/182577965-70d0e01b-0938-491e-a1eb-bc7a3c966b4d.png">